examples/file-feed/complete/src/browser-app/App.js

```diff
@@ -0,0 +1,177 @@
+import React, { Component } from 'react'
+import Dropzone from 'react-dropzone'
+import DataStore from './DataStore'
+import Feed from './Feed'
+import Preview from './Preview'
+import Peers from './Peers'
+import Status from './Status'
+import { isMediaFile } from './utils'
+import './App.css'
+
+let dataStore
+
+class App extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      feed: null,
+      files: [],
+      status: null,
+      preview: null,
+      dragActive: false,
+      showPeers: false
+    }
+  }
+
+  componentDidMount () {
+    // Set UI status message
+    this.setStatus('Starting IPFS')
+
+    // Initialize our DataStore, ie. start IPFS
+    dataStore = DataStore.init({
+      path: '/ipfs/' + new Date().toString(),
+      signalAddr: '/dns4/star-signal.cloud.ipfs.team'
+    })
+
+    dataStore.on('error', (e) => console.error(e))
+    dataStore.on('ready', () => {
```

Comment:
`on` is fine in this case, DataStore is a singleton (ie. there's only one instance of it throughout the app).